.github/workflows/nuget-publish.yml

```diff
@@ -0,0 +1,70 @@
+name: NuGet Publish
+
+on:
+  push:
+    branches: [ master ]
+
+env:
+  NUGET_DIR: ${{ github.workspace }}/nuget
+  NUGET_ARTIFACT: nuget
+
+jobs:
+  testing:
+    if: contains(github.event.head_commit.message, 'chore(release)') 
+    uses: ./.github/workflows/test.yml
+
+  pack: 
+    if: contains(github.event.head_commit.message, 'chore(release)') 
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v3
+
+      - run: sed -E 's:RepositoryCommit>.+<\/:RepositoryCommit>${{ github.sha }}<\/:g' ButterCMS/ButterCMS.csproj > tmp && mv tmp ButterCMS/ButterCMS.csproj
+
+      - run: dotnet pack --configuration Release --output ${{ env.NUGET_DIR }}
+
+      - uses: actions/upload-artifact@v3
```

Comment:
Let's pin `actions/upload-artifact` to as specific version as possible.

Reply:
Resolved in https://github.com/ButterCMS/buttercms-csharp/pull/40/commits/f3d2d5e3f00c7bfcdd3405cd42a7cfb56b556a32